### PR TITLE
Removes a lot of boilerplate from hydroponics apply, adds a mushroom mechanic for milk as implied by comments

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Removes a ton of boilerplate from `on_hydroponics_apply`
    - Replaces `chems.get_reagent_amount(type)` with `volume`. The former is a roundabout way of accomplishing the latter, the only thing the proc did was add un-necessary iterating and also round to the chemical quantisiation level, which wasn't really necessary as most locations rounded anyways. 

- While doing this, I saw th is comment: `Milk is good for humans, but bad for plants. The sugars cannot be used by plants, and the milk fat harms growth. Not shrooms though. I can't deal with this now...` This was super easy to just throw in so I did it. 

- Additionally, I noticed Uranium and Radium had this var, `tox_damage`, which was ... the same for both, but one used REM and one didn't, but it shouldn't have been using REM since it was multiplied by REM in the proc... so I removed the REM from the latter, making it doubly strong (0.5 -> 1). I did this just because I could use it nicely for the hydro proc but... I unno.

## Why It's Good For The Game

Makes it a bit clearer on working with hydro chems

## Changelog

:cl: Melbert
balance: Milk no longer harms the potency of mushrooms. Apparently it's good for them?
balance: Radium now does slightly more tox damage than Uranium
code: Removed a ton of boilerplate from chem hydro interactions
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
@MrMelbert this also seems like a mistake:

image

**Comment by MrMelbert** [comment]
> MrMelbert This also seems like a mistake:
> 
> image

That one is a little sus I won't touch it in this PR

**Comment by ExcessiveUseOfCobblestone** [comment]
I think the comment is wrong and the code is correct, not to say i wouldnt want to see those chems get btfod which i was looking at.

**Comment by ExcessiveUseOfCobblestone** [comment]
brain farting rn, with checktray removed is there an inherited way of btfoing micro (<1) doses?

**Comment by MrMelbert** [comment]
> brain farting rn, with checktray removed is there an inherited way of btfoing micro (<1) doses?

I moved it to here https://github.com/tgstation/tgstation/blob/4882b3f9035e12764ffb2c9279a8bd617bbefd8d/code/modules/hydroponics/hydroponics_chemreact.dm#L9-L12

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 1]
```diff
@@ -33,11 +33,8 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
 	//It has stable IN THE NAME. IT WAS MADE FOR THIS MOMENT.
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:35]:
```suggestion
//It has stable IN THE NAME. IT WAS MADE FOR THIS MOMENT.
```

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 2]
```diff
@@ -196,13 +193,10 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
 	// why, just why
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:195]:
```suggestion
// why, just why
```

**Review by san7890 — CHANGES_REQUESTED** [review]
spacing nits. some of them are adjacent to code you touched but you may as well patch it up now since no one else is gonna open up these files for the next three years

**Comment by san7890** [comment]
fuck it we assume direct control

**Pull request merged**